tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts

```diff
@@ -809,187 +809,275 @@ describe('image card', () => {
+    describe('thumb movement', () => {
+      describe('single selection', () => {
+        it('moves slider thumb to matched selected time', () => {
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 20},
+            end: null,
+          });
+          const timeSeries = [
+            {wallTime: 100, imageId: 'ImageId1', step: 10},
+            {wallTime: 101, imageId: 'ImageId2', step: 20},
+            {wallTime: 102, imageId: 'ImageId3', step: 30},
+            {wallTime: 103, imageId: 'ImageId4', step: 40},
+          ];
+          provideMockCardSeriesData(
+            selectSpy,
+            PluginType.IMAGES,
+            'card1',
+            null /* metadataOverride */,
+            timeSeries,
+            2 /* stepIndex */
+          );
+          const fixture = createImageCardContainer('card1');
+          fixture.detectChanges();
+          let slider = fixture.debugElement.query(By.css('mat-slider'));
+          expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('1');
+
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 30},
+            end: null,
+          });
+          store.refreshState();
+          fixture.detectChanges();
+
+          slider = fixture.debugElement.query(By.css('mat-slider'));
+          expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
+        });
 
-      store.overrideSelector(selectors.getMetricsSelectedTime, {
-        start: {step: 30},
-        end: null,
-      });
-      store.refreshState();
-      fixture.detectChanges();
+        it('does not move slider thumb on selected step with no image', () => {
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 15},
+            end: null,
+          });
+          const timeSeries = [
+            {wallTime: 100, imageId: 'ImageId1', step: 10},
+            {wallTime: 101, imageId: 'ImageId2', step: 20},
+            {wallTime: 102, imageId: 'ImageId3', step: 30},
+            {wallTime: 103, imageId: 'ImageId4', step: 40},
+          ];
+          provideMockCardSeriesData(
+            selectSpy,
+            PluginType.IMAGES,
+            'card1',
+            null /* metadataOverride */,
+            timeSeries,
+            2 /* stepIndex */
+          );
+          const fixture = createImageCardContainer('card1');
+          fixture.detectChanges();
+          let slider = fixture.debugElement.query(By.css('mat-slider'));
+          expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
+        });
 
-      slider = fixture.debugElement.query(By.css('mat-slider'));
-      expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
-    });
+        it('moves slider thumb to smaller closest step when they are close enough', () => {
+          store.overrideSelector(selectors.getMetricsSelectedTime, {
+            start: {step: 11},
```

Comment:
Maybe add a note somewhere explaining explicitly that 11 is close enough to 10, so we move to the corresponding step 0?